### PR TITLE
Fix global step update when the epoch is skipped

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

When the epoch is skipped (`on_train_batch_start` returns `-1`), `global_step` is still updated.

This to me is wrong, even though we have tests for the previous behavior.
Added by @ydcjeff in #3700

Part of https://github.com/PyTorchLightning/pytorch-lightning/pull/7357

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [n/a] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

File: tests/models/test_hooks.py [hunk 1]
```diff
@@ -229,26 +229,6 @@ def train_dataloader(self):
     trainer.fit(model)
 
 
-@pytest.mark.parametrize('max_epochs,batch_idx_', [(2, 5), (3, 8), (4, 12)])
```
Comment (carmocca) [inline tests/models/test_hooks.py:232]: Moved the test since it is testing more of the training loop internals than the model hook itself.

File: tests/trainer/loops/test_training_loop.py [hunk 2]
```diff
@@ -201,3 +202,23 @@ def run_training(**trainer_kwargs):
...
+    class CurrentModel(BoringModel):
+
+        def on_train_batch_start(self, batch, batch_idx, dataloader_idx):
+            if batch_idx == batch_idx_:
+                return -1
+
+    model = CurrentModel()
+    trainer = Trainer(max_epochs=max_epochs, limit_train_batches=10)
+    trainer.fit(model)
+    if batch_idx_ > trainer.num_training_batches - 1:
```
Comment (carmocca) [inline tests/trainer/loops/test_training_loop.py:219]: This if was not running previously with the previous condition

**Review by ananthsub — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
